hdeps/resolution.py

```diff
@@ -154,13 +158,17 @@ def drain(self) -> None:
 
             cur = chosen.get(name)
             with kev("find_best_compatible_version", project_name=name, req=str(req)):
-                version = find_best_compatible_version(
-                    project,
-                    req,
-                    self.env_markers,
-                    cur,
-                    self.current_version_callback,
-                )
+                try:
+                    version = find_best_compatible_version(
+                        project,
+                        req,
+                        self.env_markers,
+                        cur,
+                        self.current_version_callback,
+                    )
+                except NoMatchingRelease:
+                    LOG.info("Missing dep %s processing %s", req, name)
+                    continue
```

Comment:
Instead of this section what if we check if `project` is None when we get fut.result() and Log/continue there?